### ISSUE TITLE
problem with package 'phonenumbers'

### ISSUE THREAD

**Issue by sky-code** [Bouke/django-two-factor-auth#213]
Add 'phonenumbers' to requirements.txt
At this moment when installing django-two-factor-auth (1.6.1) on new virtual env you get these packages:
```
Babel==2.4.0
Django==1.11.2
django-formtools==2.0
django-otp==0.3.14
django-phonenumber-field==1.3.0
django-two-factor-auth==1.6.1
pytz==2017.2
qrcode==4.0.4
six==1.10.0
``` 
than you get an exception:
`ModuleNotFoundError: No module named 'phonenumbers'`
with this traceback:
```
Traceback (most recent call last):
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/django/utils/autoreload.py", line 227, in wrapper
    fn(*args, **kwargs)
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/django/core/management/commands/runserver.py", line 117, in inner_run
    autoreload.raise_last_exception()
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/django/utils/autoreload.py", line 250, in raise_last_exception
    six.reraise(*_exception)
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/django/utils/autoreload.py", line 227, in wrapper
    fn(*args, **kwargs)
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/django/apps/registry.py", line 108, in populate
    app_config.import_models()
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/django/apps/config.py", line 202, in import_models
    self.models_module = import_module(models_module_name)
  File "/Users/macbook/.pyenv/versions/3.6.1/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 978, in _gcd_import
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load
  File "<frozen importlib._bootstrap>", line 950, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/two_factor/models.py", line 12, in <module>
    from phonenumber_field.modelfields import PhoneNumberField
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/phonenumber_field/modelfields.py", line 8, in <module>
    from phonenumber_field import formfields
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/phonenumber_field/formfields.py", line 9, in <module>
    from phonenumber_field.phonenumber import to_python
  File "/Users/macbook/work/terrarium/venv/lib/python3.6/site-packages/phonenumber_field/phonenumber.py", line 5, in <module>
    import phonenumbers
ModuleNotFoundError: No module named 'phonenumbers'
```
as an result django-two-factor-auth not ready to use after installation

**Comment by Bouke** [comment]
Could you try and reinstall `django-phonenumber-field`? That package has a (conditional) dependency on `phonenumberslite`.

**Comment by sky-code** [comment]
No, i was trying to install `django-two-factor-auth` on new clean venv and they are not working after installation. Here is an question, who must fix this issue?
`django-two-factor-auth` or `django-phonenumber-field`
but at general `django-two-factor-auth` package not fully functional after installation

**Comment by moggers87** [comment]
I can't reproduce this issue with my own Python 3.6 virtualenv - `phonenumberslite` is installed as expected

**Comment by sky-code** [comment]
I have tried once again
```sh
MacBook-Pro-Macbook:temp macbook$ ~/.pyenv/versions/3.6.1/bin/python -m venv venv
MacBook-Pro-Macbook:temp macbook$ source venv/bin/activate
(venv) MacBook-Pro-Macbook:temp macbook$ pip freeze
(venv) MacBook-Pro-Macbook:temp macbook$ pip install django-two-factor-auth
Collecting django-two-factor-auth
  Using cached django_two_factor_auth-1.6.1-py2.py3-none-any.whl
Collecting django-formtools (from django-two-factor-auth)
  Using cached django_formtools-2.0-py2.py3-none-any.whl
Collecting django-otp<0.99,>=0.3.4 (from django-two-factor-auth)
  Using cached django_otp-0.3.14-py2.py3-none-any.whl
Collecting Django!=1.9.*,>=1.8 (from django-two-factor-auth)
  Using cached Django-1.11.2-py2.py3-none-any.whl
Collecting django-phonenumber-field<1.99,>=1.1.0 (from django-two-factor-auth)
Collecting qrcode<4.99,>=4.0.0 (from django-two-factor-auth)
Collecting pytz (from Django!=1.9.*,>=1.8->django-two-factor-auth)
  Using cached pytz-2017.2-py2.py3-none-any.whl
Collecting babel (from django-phonenumber-field<1.99,>=1.1.0->django-two-factor-auth)
  Using cached Babel-2.4.0-py2.py3-none-any.whl
Collecting six (from qrcode<4.99,>=4.0.0->django-two-factor-auth)
  Using cached six-1.10.0-py2.py3-none-any.whl
Installing collected packages: pytz, Django, django-formtools, django-otp, babel, django-phonenumber-field, six, qrcode, django-two-factor-auth
Successfully installed Django-1.11.2 babel-2.4.0 django-formtools-2.0 django-otp-0.3.14 django-phonenumber-field-1.3.0 django-two-factor-auth-1.6.1 pytz-2017.2 qrcode-4.0.4 six-1.10.0
(venv) MacBook-Pro-Macbook:temp macbook$ pip freeze
Babel==2.4.0
Django==1.11.2
django-formtools==2.0
django-otp==0.3.14
django-phonenumber-field==1.3.0
django-two-factor-auth==1.6.1
pytz==2017.2
qrcode==4.0.4
six==1.10.0
```

maybe this problem depends on environments, i was testing this on macOS Sierra, and my friend has the same issue on Ubuntu 16.04

**Comment by moggers87** [comment]
Did you try installing `django-phonenumber-field` in a venv by itself as @Bouke suggested?

**Comment by sky-code** [comment]
yep, but result is the same
```sh
MBP-Macbook:temp macbook$ ~/.pyenv/versions/3.6.1/bin/python -m venv venv
MBP-Macbook:temp macbook$ source venv/bin/activate
(venv) MBP-Macbook:temp macbook$ pip freeze
(venv) MBP-Macbook:temp macbook$ pip install django-phonenumber-field
Collecting django-phonenumber-field
Collecting babel (from django-phonenumber-field)
  Using cached Babel-2.4.0-py2.py3-none-any.whl
Collecting django>=1.5 (from django-phonenumber-field)
  Using cached Django-1.11.2-py2.py3-none-any.whl
Collecting pytz>=0a (from babel->django-phonenumber-field)
  Using cached pytz-2017.2-py2.py3-none-any.whl
Installing collected packages: pytz, babel, django, django-phonenumber-field
Successfully installed babel-2.4.0 django-1.11.2 django-phonenumber-field-1.3.0 pytz-2017.2
(venv) MBP-Macbook:temp macbook$ pip freeze
Babel==2.4.0
Django==1.11.2
django-phonenumber-field==1.3.0
pytz==2017.2
```

**Comment by Bouke** [comment]
Very strange. Works on my machine:

```
~/Developer ❯❯❯ python -m venv issue-213
~/Developer ❯❯❯ source issue-213/bin/activate
(issue-213) ~/Developer ❯❯❯ pip freeze
(issue-213) ~/Developer ❯❯❯ python -V
Python 3.6.1
(issue-213) ~/Developer ❯❯❯ pip -V
pip 9.0.1 from /Users/bouke/Developer/issue-213/lib/python3.6/site-packages (python 3.6)
(issue-213) ~/Developer ❯❯❯
(issue-213) ~/Developer ❯❯❯ pip install django-phonenumber-field
Collecting django-phonenumber-field
  Downloading django-phonenumber-field-1.3.0.tar.gz
Collecting django>=1.5 (from django-phonenumber-field)
  Downloading Django-1.11.2-py2.py3-none-any.whl (6.9MB)
    100% |████████████████████████████████| 7.0MB 202kB/s
Collecting babel (from django-phonenumber-field)
  Downloading Babel-2.4.0-py2.py3-none-any.whl (6.8MB)
    100% |████████████████████████████████| 6.8MB 198kB/s
Collecting phonenumberslite>=7.0.2 (from django-phonenumber-field)
  Downloading phonenumberslite-8.5.1-py2.py3-none-any.whl (2.7MB)
    100% |████████████████████████████████| 2.7MB 468kB/s
Collecting pytz (from django>=1.5->django-phonenumber-field)
  Using cached pytz-2017.2-py2.py3-none-any.whl
Installing collected packages: pytz, django, babel, phonenumberslite, django-phonenumber-field
  Running setup.py install for django-phonenumber-field ... done
Successfully installed babel-2.4.0 django-1.11.2 django-phonenumber-field-1.3.0 phonenumberslite-8.5.1 pytz-2017.2
(issue-213) ~/Developer ❯❯❯ pip freeze
Babel==2.4.0
Django==1.11.2
django-phonenumber-field==1.3.0
phonenumberslite==8.5.1
pytz==2017.2
```

This package depends on `django-phonenumber-field`, which depends on `phonenumberslite`. Maybe it is due to the way how `django-phonenumber-field` determines if `phonenumberslite` should be installed. Such conditional logic in `setup.py` requires that `setup.py` be executed as part of the installation. It should not use a wheel or something similar. Are you using the default pypi repository? And what happens when you clear your cache?

**Comment by sky-code** [comment]
thanks, removing cache solve the problem
```sh
MBP-Macbook:temp macbook$ ~/.pyenv/versions/3.6.1/bin/python -m venv venv2
MBP-Macbook:temp macbook$ source venv2/bin/activate
(venv2) MBP-Macbook:temp macbook$ pip install django-two-factor-auth --no-cache-dir
Collecting django-two-factor-auth
  Downloading django_two_factor_auth-1.6.1-py2.py3-none-any.whl (220kB)
    100% |████████████████████████████████| 225kB 3.0MB/s 
Collecting django-phonenumber-field<1.99,>=1.1.0 (from django-two-factor-auth)
  Downloading django-phonenumber-field-1.3.0.tar.gz
Collecting django-formtools (from django-two-factor-auth)
  Downloading django_formtools-2.0-py2.py3-none-any.whl (131kB)
    100% |████████████████████████████████| 133kB 4.6MB/s 
Collecting django-otp<0.99,>=0.3.4 (from django-two-factor-auth)
  Downloading django_otp-0.3.14-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 9.0MB/s 
Collecting Django!=1.9.*,>=1.8 (from django-two-factor-auth)
  Downloading Django-1.11.2-py2.py3-none-any.whl (6.9MB)
    100% |████████████████████████████████| 7.0MB 7.1MB/s 
Collecting qrcode<4.99,>=4.0.0 (from django-two-factor-auth)
  Downloading qrcode-4.0.4.tar.gz
Collecting babel (from django-phonenumber-field<1.99,>=1.1.0->django-two-factor-auth)
  Downloading Babel-2.4.0-py2.py3-none-any.whl (6.8MB)
    100% |████████████████████████████████| 6.8MB 8.0MB/s 
Collecting phonenumberslite>=7.0.2 (from django-phonenumber-field<1.99,>=1.1.0->django-two-factor-auth)
  Downloading phonenumberslite-8.5.1-py2.py3-none-any.whl (2.7MB)
    100% |████████████████████████████████| 2.7MB 8.5MB/s 
Collecting pytz (from Django!=1.9.*,>=1.8->django-two-factor-auth)
  Downloading pytz-2017.2-py2.py3-none-any.whl (484kB)
    100% |████████████████████████████████| 491kB 7.0MB/s 
Collecting six (from qrcode<4.99,>=4.0.0->django-two-factor-auth)
  Downloading six-1.10.0-py2.py3-none-any.whl
Installing collected packages: pytz, Django, babel, phonenumberslite, django-phonenumber-field, django-formtools, django-otp, six, qrcode, django-two-factor-auth
  Running setup.py install for django-phonenumber-field ... done
  Running setup.py install for qrcode ... done
Successfully installed Django-1.11.2 babel-2.4.0 django-formtools-2.0 django-otp-0.3.14 django-phonenumber-field-1.3.0 django-two-factor-auth-1.6.1 phonenumberslite-8.5.1 pytz-2017.2 qrcode-4.0.4 six-1.10.0
(venv2) MBP-Macbook:temp macbook$ pip freeze
Babel==2.4.0
Django==1.11.2
django-formtools==2.0
django-otp==0.3.14
django-phonenumber-field==1.3.0
django-two-factor-auth==1.6.1
phonenumberslite==8.5.1
pytz==2017.2
qrcode==4.0.4
six==1.10.0
```

**Comment by PySimonOjok** [comment]
Am trying to use this on my app but am getting the following problem that module not found, what can i do to fix this

File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "E:\ProjectMay\getA_Library\accounts\models.py", line 7, in <module>
    from university.models import *
  File "E:\ProjectMay\getA_Library\university\models.py", line 5, in <module>
    from phonenumber_field.modelfields import PhoneNumberField
ModuleNotFoundError: No module named 'phonenumber_field'

**Comment by ggjersund** [comment]
Fix: The application [django-phonenumber-field]( https://github.com/stefanfoulis/django-phonenumber-field) now requires the user to SPECIFICALLY install one of the following dependencies: `phonenumbers` or `phonenumberslite`. 

Check out [this issue](https://github.com/stefanfoulis/django-phonenumber-field/issues/326).

TLDR; `pipenv install phonenumbers`
